### ISSUE TITLE
Cannot put two containers on a web page

### ISSUE THREAD

**Issue by AlbertHart** [opensheetmusicdisplay/opensheetmusicdisplay#594]
I tried putting two containers on the same web page, and rendering part of the score to each.

This is for a project I am working on to load a score, divide it onto two pages if needed (between staves) and also add a chord only (no notes) display for comping. Which I am able to get to work. (see image below). This feature works great to create printable lead sheets from scores all in a single step. (It calculates the number of measure in the first container to it prints on a single page)

When I rendered to the second container, it turns out the render() created an identical <svg> for the second container, and wrote the score to both the old and new svg.

I was able to beat this by storing the innerHTML of the first SVG before rendering the second container, and the replacing the innerHTML of the first SVG after the second rendering.

Is this a "bug" (that only one SVG is created for two containers)?

Is there a way I could have have the second render, to a different container, use a new SVG?

![image](https://user-images.githubusercontent.com/8965352/67249467-010b5d00-f425-11e9-8824-0bb5cc1379a2.png)

**Comment by sschmidTU** [comment]
very quick reaction, the two osmd objects may share their data,
so you may have to do things in the right order.
try this:
```js
osmd1.load(...);
osmd1.render()
osmd2.load(...); // will also set options for next render of osmd1
osmd2.render();
```
see #559

**Comment by AlbertHart** [comment]
Here is what I am doing.

However, load_song() uses the same variable, osmd for both renderings. 
I'll try changing it to user a different variable for the second rendering and see what happens.

`last_measure_override = new_measure_count;
        sheet_number = 1;
        measure_start = 1;
        document.getElementById("music_container1").innerHTML = "";
        do_song();  // load XML and render to container 1

        page1_html = document.getElementById("music_container1").innerHTML
        //throw("STOP");
        height = elt.clientHeight;
        width = elt.clientWidth;
        measure_count = measures.length;
        console.log("re-render:  new width: %s new height: %s", width, height);

        
        sheet_number = 2;
        measure_start = new_measure_count + 2;
        last_measure_override = 0;
        document.getElementById("music_container2").innerHTML = "";
        do_song(); load XML and render to container 2


        // restore container 1
        document.getElementById("music_container1").innerHTML =  page1_html;
`

**Comment by AlbertHart** [comment]
I played with this a bit.

I have two functions do_song1() and do)_song2() which use separate osmds (osmd1 and osmd2) and separate containers.

When I load the page, and then run do_song1() and do_song2() manually from the javascript console. They work OK.

When i run them both from the .htm, botd containers are identical.

      ```
  do_song1();
        do_song2();
```

I then tried running them after the dom was loaded:

```
function ready() {
        console.log('DOM is ready');

        do_song1();
        do_song2();
    }

    document.addEventListener("DOMContentLoaded", ready);
```

And that didn't help.

I then inserted a 2 second delay between do_song1() and do_song2 - and it works.

So it appears the render() is asynchronous.

Perhaps we can find a way to know when it is done?

Here is the code that worked:

```
function timeFunction() {
            setTimeout(function(){ 
                do_song2();
                 }, 2000);
         }
        do_song1();
        timeFunction()
```

**Comment by sschmidTU** [comment]
how about `do_song1().then( function() { do_song2() } );`?
of course, as written in #559, ideally this shouldn't be an issue, but OSMD isn't really optimized for being rendered twice in the same container yet.

**Comment by AlbertHart** [comment]
I wasn't able to get the function().then logic to work.

But I was able to do the 4 renderings (to separate containers) to work with Promise() delay logic.

   ```
 delay = ms => new Promise(res => setTimeout(res, ms));
     
    do_all_songs = async () => {

        do_song(1);
    
        await delay(2000);
 

        if (draw_page2)
        {

            do_song(2);

            await delay(2000);

        }

        do_song(3);

        await delay(2000);

        do_song(4);

        await delay(2000);

        // restore content of the first container which is being reused,
        document.getElementById("music_container1").innerHTML =  page1_html;
    };

If you have better code for function().then logic, please pass it along.

```

**Comment by sschmidTU** [comment]
I suppose this is now also fixed with #559 and #608 being solved?
If not, i'll re-open this of course.
